Expose module method in ExportedProgram

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105575

tugsbayasgalan committed:
Expose module property in ExportedProgram

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Expose module property in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Expose module property in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Expose module method in ExportedProgram"




[ghstack-poisoned]

zhxchen17 commented on torch/_export/exported_program.py:
```diff
@@ -173,6 +313,17 @@ def __str__(self) -> str:
     def graph(self):
         return self.graph_module.graph
 
+    def module(self, flat=True) -> Optional[torch.fx.GraphModule]:
```
Let's start without having default value for flat?

tugsbayasgalan replied:
Yep good idea

tugsbayasgalan committed:
Update on "[WIP] Expose module method in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan committed:
Update on "[WIP] Expose module method in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D47734790).

tugsbayasgalan committed:
Update on "[WIP] Expose module method in ExportedProgram"


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D47734790](https://our.internmc.facebook.com/intern/diff/D47734790)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D47734790).

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D47734790).

zhxchen17 commented on torch/_export/exported_program.py:
```diff
@@ -174,6 +313,14 @@ def __str__(self) -> str:
     def graph(self):
         return self.graph_module.graph
 
+    def module(self) -> Optional[torch.fx.GraphModule]:
```
```suggestion
    def module(self) -> torch.nn.Module:
```

tugsbayasgalan committed:
Update on "Expose module method in ExportedProgram"


Differential Revision: [](https://our.internmc.facebook.com/intern/diff/)

Differential Revision: [D47734790](https://our.internmc.facebook.com/intern/diff/D47734790)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Expose module method in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge

tugsbayasgalan commented:
@pytorchbot label "topic: not user facing"

tugsbayasgalan commented:
@pytorchbot merge

tugsbayasgalan committed:
Update on "Expose module method in ExportedProgram"




[ghstack-poisoned]

tugsbayasgalan commented:
@pytorchbot merge -f "The previous tests before fixing merge conflict in test file were passing. So I think it is fine to force merge"
